pkg/apiserver/apiserver.go

```diff
@@ -166,66 +167,77 @@ func getWebService() *restful.WebService {
 func getVersions(request *restful.Request, response *restful.Response) {
 	resourceName := request.PathParameter("resource-id")
 	requestPath := request.Request.URL.Path
-	//fmt.Printf("Printing Provenance\n")
-	//provenance.TotalClusterProvenance.PrintProvenance()
-
-	// Path looks as follows:
-	// /apis/kubeprovenance.cloudark.io/v1/namespaces/default/deployments/dep1/compositions
 	resourcePathSlice := strings.Split(requestPath, "/")
 	resourceKind := resourcePathSlice[6] // Kind is 7th element in the slice
-	//provenanceInfo := provenance.TotalClusterProvenance.GetProvenance(resourceKind, resourceName)
 	provenanceInfo := "Resource Name:" + resourceName + " Resource Kind:" + resourceKind
 	fmt.Println(provenanceInfo)
 
 	response.Write([]byte(provenanceInfo))
+	intendedProvObj := provenance.FindProvenanceObjectByName(resourceName, provenance.AllProvenanceObjects)
+
+	//TODO: Validate request based on the correct namespace and the correct plural type. right now
+	//deployments is sort of just ignored intentionally. I have the namespace/pluralkind data
```

Comment:
I think if you add Postgreses to the kind_compositions.yaml then this should be fixed quickly.